src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs

```diff
@@ -92,6 +92,7 @@ private Uri GetUriForRedirect(Uri requestUri, HttpResponseMessage response)
                 case HttpStatusCode.SeeOther:
                 case HttpStatusCode.TemporaryRedirect:
                 case HttpStatusCode.MultipleChoices:
+                case HttpStatusCode.PermanentRedirect:
```

Comment:
What about https://github.com/dotnet/corefx/blob/015ee21f6fe1eadad478d8c9409c95cf98c2b121/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs#L652-L667 ?
cc: @vancem

Reply:
This will probably need to be updated too. It looks like the test for `MaxAutomaticRedirections` only uses 302, so I'll add an additional test for that case.
https://github.com/dotnet/corefx/blob/909dc9eb3ba829647c45348c85bc9271019c509f/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs#L1083-L1132

Reply:
I added the new redirect code here, but it is causing [build failures on netfx](https://ci3.dot.net/job/dotnet_corefx/job/master/job/windows-TGroup_netfx+CGroup_Debug+AGroup_x86+TestOuter_true_prtest/94/console). I can use the integer 308 instead, or do you think I should take a different approach?

Reply:
Using "308" might still fail on NETFX since it might not be supported.  You can try though using the integer. But if the test fails, then you'll need to skip that test case on NETFX.

Reply:
NETFX does NOT support 308 redirect code:

```c#
                //
                // Check for Redirection
                //
                // Table View:
                // Method            301             302             303             307
                //    *                *               *             GET               *
                // POST              GET             GET             GET            POST
                //
                // Put another way:
                //  301 & 302  - All methods are redirected to the same method but POST. POST is redirected to a GET.
                //  303 - All methods are redirected to GET
                //  307 - All methods are redirected to the same method.
                //
                else if (AllowAutoRedirect && (
                    ResponseStatusCode==HttpStatusCode.Ambiguous          || // 300
                    ResponseStatusCode==HttpStatusCode.Moved              || // 301
                    ResponseStatusCode==HttpStatusCode.Redirect           || // 302
                    ResponseStatusCode==HttpStatusCode.RedirectMethod     || // 303
                    ResponseStatusCode==HttpStatusCode.RedirectKeepVerb ))   // 307
```